### PR TITLE
Fix python3 six and make issue on Amazon Linux 2

### PR DESCRIPTION
**Pull request by shahmishal**
```
---------------------------------------------
Error when bootstrapping CMake:
Cannot find appropriate Makefile processor on this system.
Please specify one using environment variable MAKE.
---------------------------------------------
```

### CODE REVIEW

File: platforms/Linux/amazonlinux/2/build_rpm.sh [hunk 1]
```diff
@@ -32,6 +32,9 @@ cp patches/*.patch $HOME/rpmbuild/SOURCES/
 pushd $HOME/rpmbuild/SPECS
 # install all the dependencies needed to build Swift from the spec file itself
 yum-builddep -y ./swiftlang.spec
+# fix python six
+mkdir -p /usr/local/lib/python3.7/site-packages/
+easy_install-3.7 six
```
[the remark this answers is not in the document]
Comment (shahmishal) [reply platforms/Linux/amazonlinux/2/build_rpm.sh:37]: Yes, we should drop CMake from BuildRequires. I will create a new PR for that

**Comment by compnerd** [comment]
> The `Requires` list should only list the additional packages (dependencies) that need to be installed. e.g.
> 
> ```
> Requires
> binutils
> gcc
> git
> glibc-static
> gzip
> libbsd
> libcurl
> libedit
> libicu
> libsqlite3x
> libstdc++-static
> libuuid
> libxml2
> python3
> tar
> tzdata
> ```

I think that gcc should be a `BuildRequires` only.  Additionally, the `libstdc++-static` and `glibc-static` should be replaced with `libstdc++` and `glibc`.  The runtime dynamically links against the C/C++ runtime (and must in order to avoid any ODR).  `libbsd` has not been needed for a while and should be dropped.  `libicu` seems suspect - we provide our own ICU library builds.  I wonder if we can tweak the build to statically link against `sqlite`, `libxml2`, `uuid`, and `libcurl`.

I think that we are including more runtimes than necessary (e.g. hwasan), and simultaneously under including files (e.g. the profile runtime).

**Comment by tomerd** [comment]
> I think that gcc should be a BuildRequires only. Additionally, the libstdc++-static and glibc-static should be replaced with libstdc++ and glibc. The runtime dynamically links against the C/C++ runtime (and must in order to avoid any ODR). libbsd has not been needed for a while and should be dropped. libicu seems suspect - we provide our own ICU library builds. I wonder if we can tweak the build to statically link against sqlite, libxml2, uuid, and libcurl.

> I think that we are including more runtimes than necessary (e.g. hwasan), and simultaneously under including files (e.g. the profile runtime).

this list comes from our existing docker setup. iirc the `-static` variants are needed for static linking. @drexin?

**Comment by compnerd** [comment]
Just to verify, cmake is being dropped because we build it and use it?  If so, doing the rest subsequently SGTM.

**Comment by shahmishal** [comment]
Correct, we are building CMake to make sure we have the latest version required for Swift build system.

**Comment by tachoknight** [comment]
> Correct, we are building CMake to make sure we have the latest version required for Swift build system.

Is this a new change in main? I've modified my spec file to build ninja so that BuildDependency is gone, but I don't see a recipe in the `build-presets.ini` that also builds CMake as well.

**Comment by shahmishal** [comment]
This was added ~2 years back by @compnerd - https://github.com/apple/swift/pull/26301

**Comment by shahmishal** [comment]
It would be a good to review all of the requirements in Dockerfiles and RPM spec file. Created an issue - https://bugs.swift.org/browse/SR-15418 to track the work.

**Pull request merged**